File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <stdbool.h>
#include <math.h>

bool is_armstrong_number(int candidate) {
    int number = candidate;
    
    int length = (int)floor(log10((float)number)) + 1;
    int arr[length];
    for(int x = 0; number != 0; x++) {
        arr[x] = number % 10;
        number /= 10;
    } 

    int output = 0;
    for(int x = 0; x < length; x++) {
        output += pow(arr[x], length);
    }

    return (output == candidate);
}

File: solutions/java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.lang.Math;

class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int[] digits = IntToArray(numberToCheck);

        int digitSum = 0;
        int exp = digits.length;

        for (int x = 0; x < digits.length; x++) {
            digitSum += (int) Math.pow(digits[x], exp);
        }

        return digitSum == numberToCheck;
    }

    static int[] IntToArray(int n) {
        int index = 0;
        int len = Integer.toString(n).length();
        int[] arr = new int[len];

        while(n != 0) {
            arr[len - index - 1] = n % 10;
            n = n / 10;
            index++;
        }
        return arr;
    }
}
